### PR TITLE
Fix issue with directory traversal check being too restrictive.

### PR DESCRIPTION
**Pull request by gramk**
Add tests for cache assets.

### CODE REVIEW

File: src/test/scala/smoke/StaticAssetsTest.scala [hunk 1]
```diff
@@ -53,6 +55,68 @@ class StaticAssetsTest extends FunSpecLike {
...
+      val pw2 = new PrintWriter(file)
+      pw2.write("after-cache")
+      pw2.close
+
+      val response2 = staticAssets.responseFromAsset("/cached.txt")
+      assert(response2.status === Ok)
+      assert(response2.body.asInstanceOf[RawData].data === "before-cache".getBytes)
+    }
+
+    it("should not list assets when asking for a folder, when publis is in a jar file") {
```
Comment (damienlevin) [inline src/test/scala/smoke/StaticAssetsTest.scala]: Typo

File: src/test/scala/smoke/StaticAssetsTest.scala [hunk 2]
```diff
@@ -53,6 +55,68 @@ class StaticAssetsTest extends FunSpecLike {
...
+      val staticAssets = new MockAsset(false)
+      val response = staticAssets.responseFromAsset("/")
+      assert(response.status === NotFound)
+    }
+
+    ignore("should not prevent from loading resources with sbt run") {}
```
Comment (damienlevin) [inline src/test/scala/smoke/StaticAssetsTest.scala]: To be removed

File: src/main/resources/public/index.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<h1>Hello</h1>
```
Comment (damienlevin) [inline src/main/resources/public/index.html]: Do we need that file ?

File: src/test/scala/smoke/StaticAssetsTest.scala (hunk 2 above)
Comment (gramk) [inline src/test/scala/smoke/StaticAssetsTest.scala]: It's there because it needs to be tested but I didn't find a way yet.

**Pull request merged**